File: collector/src/main/java/com/bitmovin/analytics/utils/DataSerializer.java
package com.bitmovin.analytics.utils;

import com.google.gson.Gson;
import com.google.gson.JsonSyntaxException;

public class DataSerializer {
    public static <T> String serialize(T data) {
        Gson gson = new Gson();
        return gson.toJson(data);
    }

    public static <T> T deserialize(String input, Class<T> classOfT) {
        T response = null;
        try {
            response = new Gson().fromJson(input, classOfT);
            return response;
        } catch (JsonSyntaxException e) {
            return response;
        }

    }
}

File: collector/src/main/java/com/bitmovin/analytics/utils/DataSerializer.kt
package com.bitmovin.analytics.utils

import com.google.gson.Gson
import com.google.gson.JsonSyntaxException

object DataSerializer {

    fun <T> serialize(data: T?): String? {
        return data?.let { Gson().toJson(it) }
    }

    fun <T> trySerialize(data: T?): String? {
        return try {
            serialize(data)
        } catch (ignored: Exception) { null }
    }

    fun <T> deserialize(input: String?, classOfT: Class<T>?): T? {
        return input?.let {
            return try {
                Gson().fromJson(input, classOfT)
            } catch (e: JsonSyntaxException) {
                null
            }
        }
    }
}
